src/tests/ftest/control/daos_agent_support_collect_log.py

```diff
@@ -0,0 +1,53 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from support_test_base import SupportTestBase
+
+
+class DaosAgentSupportCollectLogTest(SupportTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+            Verify the daos_server support collect-log command.
+
+    :avocado: recursive
+    """
+
+    def test_daos_agent_collect_log(self):
+        """JIRA ID: DAOS-10625
+
+        Test Description:
+            Test daos_agent support collect-log command completes successfully.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=control,basic,support,daos_server
+        :avocado: tags=test_daos_agent_support_collect_log
+        """
+        # Create the custom log data which will be collected via support collect-log,
+        # Later verify the data file is archived as part of collection.
+        self.log_hosts = self.hostlist_clients
+        self.create_custom_log("Client_Support_Logs")
+
+        # Run daos_agent support collect-log with --extra-logs-dir,
+        # copy log to folder with command option --target-folder
+        # Enable archive mode.
+        result = self.agent_managers[0].support_collect_log(
+            extra_logs_dir=self.custom_log_dir,
+            target_folder=self.target_folder,
+            archive=True)
+
+        if not result.passed:
+            self.fail("Failed to run daos_agent support collect-log command")
+
+        # Extract the collected tar.gz file
+        result = self.extract_logs(self.target_folder + ".tar.gz")
+        if result is not None:
+            self.fail(result)
+
+        # Verify the custom log file collected on each clients.
+        self.verify_custom_log_data()
+
+        # Clean up the log file created during test execution
+        self.cleanup_support_log(self.target_folder)
```

Comment:
This won't run if the test fails. I'll let @phender (more familiar) comment on this

Reply:
We could follow this pattern:
https://github.com/daos-stack/daos/blob/652fff38ef0677303669ee9c840f77279027f65a/src/tests/ftest/util/nvme_utils.py#L68
https://github.com/daos-stack/daos/blob/652fff38ef0677303669ee9c840f77279027f65a/src/tests/ftest/util/nvme_utils.py#L79-L97

By changing `self.cleanup_support_log` to return a list of errors instead of calling `self.fail`, and then right after calling `support_collect_log`, register the cleanup with:
`self.register_cleanup(self.reset_fault_device, log_dir=self.target_folder)`

This cleanup will be ran in teardown, which accounts for test failures and timeouts